packages/protocol/contracts/governance/VestingInstance.sol

```diff
@@ -0,0 +1,379 @@
+    bool vestingRevocable,
+    address vestingRevoker,
+    address vestingRefundDestination,
+    address registryAddress
+  ) public {
+    // perform checks on the input data
+    require(registryAddress != address(0), "registry address must not be the zero address");
+    require(vestingAmount > 0, "Vesting amount must be positive");
+    require(vestingBeneficiary != address(0), "Beneficiary is the zero address");
+    require(vestingRefundDestination != address(0), "Refund destination is the zero address");
+    require(vestingCliff <= vestingPeriodSec, "Vesting cliff is longer than vesting duration");
+    require(vestingPeriodSec > 0, "Vesting period is 0 s.");
+    require(
+      vestAmountPerPeriod <= vestingAmount,
+      "Vesting amount per period is greater than the total vesting amount"
+    );
+    require(
+      vestingStartTime.add(vestingCliff) > block.timestamp,
+      "Vesting end time is before current time"
+    );
+
+    // set the registry address
+    setRegistry(registryAddress);
+
+    // init the vesting scheme
+    vestingScheme.vestingPeriods = vestingAmount.div(vestAmountPerPeriod);
+    vestingScheme.vestingAmount = vestingAmount;
+    vestingScheme.vestAmountPerPeriod = vestAmountPerPeriod;
+    vestingScheme.vestingPeriodSec = vestingPeriodSec;
+    vestingScheme.vestingCliffStartTime = vestingStartTime.add(vestingCliff);
+    vestingScheme.vestingStartTime = vestingStartTime;
+
+    // init the state vars
+    beneficiary = vestingBeneficiary;
+    revocable = vestingRevocable;
+    refundDestination = vestingRefundDestination;
+    revoker = vestingRevoker;
+  }
+
+  /**
+   * @notice Transfers available released tokens from the vesting back to beneficiary.
+   */
+  function withdraw() external nonReentrant onlyBeneficiary {
+    bool isUnpaused = !paused || (paused && block.timestamp >= pauseEndTime);
+    require(isUnpaused, "Withdrawals only allowed in the unpaused state");
+    if (isUnpaused) paused = false;
+    uint256 withdrawableAmount = getWithdrawableAmountAtTimestamp(block.timestamp);
+    require(withdrawableAmount > 0, "No gold is due for withdrawal");
+    currentlyWithdrawn = currentlyWithdrawn.add(withdrawableAmount);
+    require(getGoldToken().transfer(beneficiary, withdrawableAmount), "Withdrawal of gold failed");
+  }
+
+  /**
+   * @notice Allows only the revoker to revoke the vesting. Gold already vested
+   * remains in the contract, the rest is returned to the _refundDestination.
+   * @param revokeTimestamp the revocation timestamp
+   * @dev If revokeTimestamp is less than the current block timestamp, it is set equal to the latter
+   */
+  function revoke(uint256 revokeTimestamp) external nonReentrant onlyRevoker {
+    require(revocable, "Revoking is not allowed");
+    require(!revoked, "Vesting already revoked");
+    revokeTime = revokeTimestamp > block.timestamp ? revokeTimestamp : block.timestamp;
+    uint256 balance = getGoldToken().balanceOf(address(this));
```

Comment:
Same question as below. Maybe we need a helper function that looks up your balance, which is the sum of the balance of this contract and your locked gold balance?

Reply:
helper function added